woocommerce/woocommerce-gateway-stripe
# Clear webhook state after reconfiguring webhooks

Fixes #3280 

## Changes proposed in this Pull Request:

When a user reconfigures their webhook, clear the webhook state. The webhook state stores errors and success statuses of webhook processing and so after reconfiguring the webhooks, it's no longer relevant and so should be cleared. 

## Testing instructions

1. Set up webhooks. Make sure your site is publically accessible. 
2. In your Stripe dashboard duplicate your webhook endpoint so you have 2 endpoints set up.
   - This will cause one of the webhooks to fail because the key will invalidate. 

<p align="center">
<img width="1000" alt="Screenshot 2024-08-12 at 12 48 33 PM" src="https://github.com/user-attachments/assets/ca359185-b8ee-4ee9-9e60-d3e6482be5d0">
</br>
<sup>Duplicate webhooks sent to my local site.</sup>
</p> 

3. Trigger a webhook event. eg purchase a product. 
4. Go to you're Stripe plugin settings. You'll notice that there's a webhook warning. 

<p align="center">
<img width="500" alt="Screenshot 2024-08-12 at 12 51 30 PM" src="https://github.com/user-attachments/assets/56eaef03-3398-4410-b4dc-d9d73ec0c6de">
</p> 

5. Open the connection modal by clicking the **Configure connection**
6. Click the **Reconfigure webhooks** button.
7. Refresh the settings page. 
   - On develop the webhook warning will still be shown. 
   - On this branch, the webhook state would have been cleared and so the warning should disappear. 







---


**Post merge**
